### PR TITLE
spykfunc: provide alternative deployment w/ spark 3.0

### PR DESCRIPTION
**Pull request by matz-e**

### CODE REVIEW

File: deploy/environments/applications.yaml [hunk 1]
```diff
@@ -67,6 +67,7 @@ spack:
     - reportinglib%intel
     - spatial-index
     - spykfunc
+    - spykfunc@0.16.0 ^spark@3.0.0 ^py-pyspark@3.0.0
```
Comment (pramodk) [inline deploy/environments/applications.yaml:70]: should this version go into production as well? i.e. if 3.0 is only stable and we have issues with newer version, we can just provide single module for end users?

Comment (matz-e) [reply deploy/environments/applications.yaml:70]: I got stuff to work at scale with 3.1.1 as well. Seeing as it is hard to debug the interaction between Python/JVM/C++, I'd rather stay with the newer versions. Reading through the changelog, there seem to be tangential fixes to features we're using.

I've got to tune this a little, anyways.

Comment (pramodk) [reply deploy/environments/applications.yaml:70]: Ok. But you also keep `3.0` based module? `^spark@3.0.0: '{name}/{version}-spark3.0'`

Comment (matz-e) [reply deploy/environments/applications.yaml:70]: Two modules, one for 3.1 by default, one 3.0

Comment (mgeplf) [reply deploy/environments/applications.yaml:70]: @matz-e / @pramodk: is the idea is to have both avaiable, in case one doesn't end up being reliable?

Comment (matz-e) [reply deploy/environments/applications.yaml:70]: Yes. From the Spark side, prior executions with 3.1 went fine, so this is really just a precautionary measure.

**Review by mgeplf — APPROVED** [review]

**Pull request merged**